helix-lock/src/main/java/org/apache/helix/lock/LockInfo.java

```diff
@@ -19,28 +19,109 @@
 
 package org.apache.helix.lock;
 
-import java.util.Map;
+import org.apache.helix.HelixProperty;
+import org.apache.helix.ZNRecord;
 
 
 /**
- * Generic interface for a map contains the Helix lock information
- * @param <T> The type of the LockInfo value
+ * Structure represents a lock node information, implemented using ZNRecord
  */
-public interface LockInfo<T> {
+public class LockInfo extends HelixProperty {
```

Comment:
Put a TODO here. This Helix Property will be replaced since distributed lock will be a module independent from Helix.

Reply:
In this case I can remove the "extends HelixProperty" to make it independent from Helix. The LockInfo is barely a ZNRecord, and currently I'm not using any properties and methods exclusively provided by HelixProperty.

Reply:
Sounds good.